pytest-selenium/tests/test_toc.py

```diff
@@ -19,7 +21,10 @@ def test_toc_toggle_button_opens_and_closes(selenium, base_url, book_slug, page_
     """
     # GIVEN: The selenium driver, base_url, book_slug, and page_slug
 
-    # WHEN: The book and page URL is loaded
+    # WHEN: The page is loaded
+    book_list = utility.Library()
```

Comment:
This would be better as a fixture. A suggestion for this test would be.

`def test_toc_toggle_button_opens_and_closes(selenium, base_url, random_book_slug, page_slug):`

You could implement `random_book_slug` as a fixture which would do these same actions and return what is needed. In other words, what is happening here is really a `GIVEN` and involves setting up the test. In most cases this can be taken care of by a fixture. You would place this in `conftest.py` and might look something like this:

```
@pytest.fixture
def random_book_slug():
    book_list = utility.Library()
    return book_list.random_book_slug()
```
A good reference for fixtures is the pytest documentation: http://doc.pytest.org/en/latest/fixture.html

Let me know if you want more help with this.

Reply:
Thanks for the suggestion @m1yag1 . It works and I pushed the changes